src/App/App.xaml.cs

```diff
@@ -155,6 +157,12 @@ private async Task CheckPasswordlessLoginRequestsAsync()
                 return;
             }
 
+            // Wait for the app to be in the foreground to show the pop up modal
+            while (!_isResumed)
+            {
+                await Task.Delay(500);
+            }
```

Comment:
I'd like to avoid having this Delay thing, what do you think of having something like using a flag to indicate that there is a pending operation to check the passwordless login request and trigger this method again?

```c#
// maybe it needs to be volatile, I'm not sure if there could be a thread update issue, not at first sight but wanted to point it out just in case
private bool _pendingCheckPasswordlessLoginRequests;

...

private async Task CheckPasswordlessLoginRequestsAsync()
{
    if(!_isResumed)
    {
        return;
    }

    _pendingCheckPasswordlessLoginRequests = false;

    if (await _vaultTimeoutService.IsLockedAsync())
    {
        return;
    }

...

else if (message.Command == "passwordlessLoginRequest" || message.Command == "unlocked")
{
    _pendingCheckPasswordlessLoginRequests = true;
    CheckPasswordlessLoginRequestsAsync().FireAndForget();
}

...

protected async override void OnStart()
{
    System.Diagnostics.Debug.WriteLine("XF App: OnStart");
    _isResumed = true;
    if(_pendingCheckPasswordlessLoginRequests)
    {
        CheckPasswordlessLoginRequestsAsync().FireAndForget();
    }

...

protected override void OnResume()
{
    System.Diagnostics.Debug.WriteLine("XF App: OnResume");
    _isResumed = true;
    // dunno if it's necessary here
    if(_pendingCheckPasswordlessLoginRequests)
    {
        CheckPasswordlessLoginRequestsAsync().FireAndForget();
    }
```